core/access_tokens.go

```diff
@@ -87,6 +88,15 @@ func (a *API) deleteAccessToken(ctx context.Context, x struct{ ID string }) erro
 	if currentID == x.ID {
 		return errCurrentToken
 	}
-	return a.accessTokens.Delete(ctx, x.ID)
-	// TODO(tessr): delete any associated grants, as well
+	err := a.accessTokens.Delete(ctx, x.ID)
+	if err != nil {
+		return err
+	}
+
+	err = a.revokeGrantsByAccessToken(ctx, x.ID)
+	if err != nil {
+		// well, technically we did delete the access token, so don't return the error
+		log.Printkv(ctx, log.KeyError, "revoking grants for access token %s", x.ID)
+	}
```

Comment:
it's not _that_ important, but I suppose we could return an error. it would be weird to return an error if the token actually got deleted though (and we have no concept of, like, database transactions across raftdb and postgres).